### ISSUE TITLE
timeout setting is not passed to job submission

### ISSUE THREAD

**Issue by chunfuchen** [Qiskit/qiskit-terra#919]
### Informations

- **Qiskit Terra version**: the newest master branch
- **Python version**: 3.6.5
- **Operating system**: macOS 10.13

### What is the current behavior?

`timeout` setting only used for `_wait_for_result` but not `_wait_for_submission`.
So, Qiskit raises `the Timeout error` after 60 seconds regardless of the setting of `timeout`.

related codes:
`_wait_for_result`
https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/backends/ibmq/ibmqjob.py#L190
`_wait_for_submission`
https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/backends/ibmq/ibmqjob.py#L400

### What is the expected behavior?

`timeout` setting should be passed to `_wait_for_submission` as well.

### Suggested solutions

pass the `timeout`  in `_wait_for_result` to  `_wait_for_submission`.

**Comment by jaygambetta** [comment]
@delapuente can we link this to the epic on the redo of backends. 

I want to discuss with should we include these or not in the ibmqprovoder

**Comment by delapuente** [comment]
I don't think this belongs to the authentication effort but we can discuss it in the next review if you want.
